pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -141,9 +145,141 @@ func (pl *PodTopologySpread) EventsToRegister() []framework.ClusterEventWithHint
 		// an unschedulable Pod schedulable.
 		// - Delete. An unschedulable Pod may fail due to violating an existing Pod's topology spread constraints,
 		// deleting an existing Pod may make it schedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.All}},
-		// Node add|delete|updateLabel maybe lead an topology key changed,
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.All}, QueueingHintFn: pl.isSchedulableAfterPodChange},
+		// Node add|delete|update maybe lead an topology key changed,
 		// and make these pod in scheduling schedulable or unschedulable.
-		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Delete | framework.UpdateNodeLabel}},
+		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add | framework.Delete | framework.Update}, QueueingHintFn: pl.isSchedulableAfterNodeChange},
 	}
 }
+
+func shouldNotBeAffectedByPodChange(incomingPod, targetPod *v1.Pod) bool {
+	return incomingPod.Spec.NodeName == "" || incomingPod.Namespace != targetPod.Namespace
+}
+
+func (pl *PodTopologySpread) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalPod, modifiedPod, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, err
+	}
+
```

Comment:
Instead of using `involvedInTopologySpreading ` many places, I prefer using it once here and return as early as possible:
```suggestion
if !involvedInTopologySpreading(pod, modifiedPod) {
  logger.V(5).Info("the updated pod is not involved in the unscheduled pod's topology spread constraints", ...)
  return framework.QueueSkip, nil
}
```